Source/Initialization/WarpXInitData.cpp

```diff
@@ -580,24 +582,42 @@ WarpX::InitLevelData (int lev, Real /*time*/)
                                                  Ezfield_parser->compile<3>(),
                                                  m_edge_lengths[lev],
                                                  lev);
+
+#ifdef AMREX_USE_EB
+       //We initialize ECTRhofield consistently with the Efield
+        if (WarpX::maxwell_solver_id == MaxwellSolverAlgo::ECT) {
+            m_fdtd_solver_fp[lev]->EvolveECTRho(Efield_fp[lev], m_edge_lengths[lev],
+                                                m_face_areas[lev], ECTRhofield[lev], lev);
+
+        }
+#endif
+
        if (lev > 0) {
-          InitializeExternalFieldsOnGridUsingParser(Efield_aux[lev][0].get(),
-                                                    Efield_aux[lev][1].get(),
-                                                    Efield_aux[lev][2].get(),
-                                                    Exfield_parser->compile<3>(),
-                                                    Eyfield_parser->compile<3>(),
-                                                    Ezfield_parser->compile<3>(),
-                                                    m_edge_lengths[lev],
-                                                    lev);
+            InitializeExternalFieldsOnGridUsingParser(Efield_aux[lev][0].get(),
+                                                      Efield_aux[lev][1].get(),
+                                                      Efield_aux[lev][2].get(),
+                                                      Exfield_parser->compile<3>(),
+                                                      Eyfield_parser->compile<3>(),
+                                                      Ezfield_parser->compile<3>(),
+                                                      m_edge_lengths[lev],
+                                                      lev);
+
+            InitializeExternalFieldsOnGridUsingParser(Efield_cp[lev][0].get(),
+                                                      Efield_cp[lev][1].get(),
+                                                      Efield_cp[lev][2].get(),
+                                                      Exfield_parser->compile<3>(),
+                                                      Eyfield_parser->compile<3>(),
+                                                      Ezfield_parser->compile<3>(),
+                                                      m_edge_lengths[lev],
+                                                      lev);
+#ifdef AMREX_USE_EB
+           if (WarpX::maxwell_solver_id == MaxwellSolverAlgo::ECT) {
+               //We initialize ECTRhofield consistently with the Efield
+               m_fdtd_solver_fp[lev]->EvolveECTRho(Efield_cp[lev], m_edge_lengths[lev],
```

Comment:
Just checking, should this be `m_fdtd_solver_cp` instead of `m_fdtd_solver_fp`, since we are passing `Efield_cp` and since we seem to be inside the `if (lev > 0)` block?
```suggestion
               m_fdtd_solver_cp[lev]->EvolveECTRho(Efield_cp[lev], m_edge_lengths[lev],
```

Reply:
Good catch! On the other hand this is there for when we'll implement EB+mesh refinement, which is not there yet (as you see m_edge_lengths and m_face areas etc. only exist on the fine patch..)